src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -164,10 +164,19 @@ UpdateTaskQueryString(Query *query, Oid distributedTableId, RangeTblEntry *value
 
 	if (valuesRTE != NULL)
 	{
+		RangeTblEntry *distributedRte = linitial(query->rtable);
+
 		Assert(valuesRTE->rtekind == RTE_VALUES);
 
 		oldValuesLists = valuesRTE->values_lists;
 		valuesRTE->values_lists = task->rowValuesLists;
+
+		/* setting an alias simplifies deparsing of RETURNING */
+		if (distributedRte->alias == NULL)
+		{
+			Alias *alias = makeAlias(CITUS_TABLE_ALIAS, NIL);
+			distributedRte->alias = alias;
```

Comment:
For upsert and INSERT..SELECT we set the alias in `RebuildQueryStrings` before calling `UpdateTaskQueryString`. Would it make sense to set the alias in `RebuildQueryStrings` for multi-row INSERTs as well? 

(we should probably clean this up and do it for all INSERTs)